sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -216,19 +216,23 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
         c
       }
 
-    case DropTable(ResolvedV1TableIdentifier(ident), ifExists, purge) =>
+    case DropTable(ResolvedV1Identifier(ident), ifExists, purge) =>
```

Comment:
I am afraid this breaks the session catalog delegation. Previously, we checked the table was `V1Table`. Right now, we simply check the identifier looks like a V1 table identifier, which still may point to a valid V2 table. If I have a custom session catalog, it may be able to load both V1 and V2 tables. After this change, the V1 drop code is invoked for V2 tables in custom session catalogs.

Reply:
@cloud-fan @viirya @dongjoon-hyun, could you double check if I missed anything?

Reply:
I checked the difference between `ResolvedV1TableIdentifier` and `ResolvedV1Identifier`. So do you mean `ResolvedV1Identifier` could wrongly apply on a V2 table? I.e.,

> case ResolvedIdentifier(catalog, ident) if isSessionCatalog(catalog)

If `catalog` is a custom session catalog which is capable for V1 and V2 tables?

Reply:
I saw for many commands, there is a `isV2Provider` check, but `DropTable` doesn't. So seems we need it?

Reply:
I think `ResolvedV1Identifier` simply means it is an identifier in the session catalog that has only db and table name. In custom session catalogs, it may point to a valid V2 table.

Reply:
Seems like @viirya is right and we need a call to `isV2Provider`.

Reply:
The change here is necessary as we don't look up the table for DROP TABLE anymore, so we can only "guess" it's a v1 or v2 table by the identifier.

I don't have a good solution here unless we add back the table lookup for DROP TABLE. Or shall we remove v1 DROP TABLE entirely? This is a very simple command and I think the v2 command covers it very well.

Reply:
I would be fine either way. I understand the goal to avoid an extra table lookup. Is the idea here that catalogs can override `tableExists` and that would be cheaper than `loadTable`? The default `tableExists` in V2 catalogs still relies on loading tables.

Reply:
The reason I asked that question is because the PR mentions that V2 tables may now skip table lookup. I am not sure how it is possible given the current `DropTableExec` implementation that calls `tableExists` before `drop`. The same is true for `DropTableCommand`. If I understand correctly, the goal of the PR is to avoid one extra lookup. Can we achieve that by referencing tables inside `DropTableExec`, instead of an identifier? I guess removing V1 DROP is also an option that would work. We may need to check `V2SessionCatalog`. It does an extra lookup in drop, which may be not needed.

I created SPARK-43203 for this.

Reply:
`exists` should be slightly faster than `load` as the RPC message is smaller. I think we can further optimize out `exists` and send one RPC to drop the table. If the table doesn't exist, false will be returned and we can deal with it in the client (Spark) side to implement the `if exists` semantic.

This PR also makes the third-party access control service clearer: ideally people don't need SELECT permission to drop a table.

Reply:
Sounds good to me. Shall we switch to V2 DROP path for all cases to fix [SPARK-43203](https://issues.apache.org/jira/browse/SPARK-43203)?

Reply:
> I think we can further optimize out exists and send one RPC to drop the table

+1 as the public catalog API already requires catalogs to return false for `dropTable` if the table does not exist

Reply:
> Shall we switch to V2 DROP path for all cases to fix [SPARK-43203](https://issues.apache.org/jira/browse/SPARK-43203)?

Yea we should. Can you create a PR? thanks!

Reply:
Sounds good to me to switch to V2 DROP.

Reply:
I'll have time, probably, on Monday.

Reply:
@aokolnychyi Any update for this? If you don't mind I can finish it this weekend.😄

Reply:
If you create a management table in spark 3.5.1, you cannot delete the path when dropping the table. I think this is a bug. It can be deleted correctly in spark 3.3.